client/components/data/query-gutenberg-create-post/index.jsx

```diff
@@ -0,0 +1,45 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { createGutenbergPostDraft } from 'state/gutenberg/actions';
+
+export class QueryGutenbergCreatePost extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+	};
+
+	componentDidMount() {
+		this.request();
+	}
+
+	componentDidUpdate( { siteId } ) {
+		if ( siteId !== this.props.siteId ) {
+			this.request();
+		}
+	}
+
+	request() {
+		const { siteId, createGutenbergPostDraft: dispatchCreateGutenbergPostDraft } = this.props;
+		if ( siteId ) {
+			dispatchCreateGutenbergPostDraft( siteId );
+		}
+	}
+
+	render() {
+		return null;
+	}
+}
+
+export default connect(
+	null,
+	{ createGutenbergPostDraft }
+)( QueryGutenbergCreatePost );
```

Comment:
this is a pretty unexpected use of a query component - is the `CreatePostDraft` a kind of resource we might imagine polling for or is it intended to be a kind of _action_ like "create this post draft"?

if it's the first would it make more sense to call it something like `<QueryPostDraft />` and if it's the second would it make sense to dispatch it as an action instead and use some declarative query component to retrieve it?

Reply:
Generally query components are reserved for GETs and intended to be idempotent (can be called multiple times without side-effects). Unless I'm misreading this, let's dispatch this in the parent component.